src/pudl/transform/ferc714.py

```diff
@@ -533,3 +542,110 @@ def out_ferc714__hourly_planning_area_demand(
         df[columns], table_name="out_ferc714__hourly_planning_area_demand"
     )
     return df
+
+
+@asset(
+    io_manager_key="pudl_io_manager",
+    compute_kind="pandas",
+)
+def core_ferc714__yearly_planning_area_demand_forecast(
+    raw_ferc714__yearly_planning_area_demand_forecast: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform the yearly planning area forecast data per Planning Area.
+
+    Transformations include:
+
+    - Drop/rename columns.
+    - Remove duplicate rows and average out the metrics.
+
+    Args:
+        raw_ferc714__yearly_planning_area_demand_forecast: Raw table containing,
+            for each year and each planning area, the forecasted summer and winter peak demand,
+            in megawatts, and annual net energy for load, in megawatthours, for the next
+            ten years.
+
+    Returns:
+        Clean(er) version of the yearly forecasted demand by Planning Area.
+    """
+    # Clean up columns
+    df = _pre_process(
+        raw_ferc714__yearly_planning_area_demand_forecast,
+        table_name="core_ferc714__yearly_planning_area_demand_forecast",
+    )
+
+    # For any rows with non-unique respondent_id_ferc714/report_year/forecast_year,
+    # group and take the mean measures
+    # For the 2006-2020 data, there were only 20 such rows. In most cases, demand metrics were identical.
+    # But for some, demand metrics were different - thus the need to take the average.
+    logger.info(
+        "Removing non-unique report rows and taking the average of non-equal metrics."
+    )
```

Comment:
I don't expect that there will be updates to the old DBF/CSV data, but I think it would be good to log the number of duplicate rows here, and assert that there aren't more than we expect to see.

Reply:
> I don't expect that there will be updates to the old DBF/CSV data, but I think it would be good to log the number of duplicate rows here, and assert that there aren't more than we expect to see.

Adding in now